VariationalBayes/ExponentialFamilies.py

```diff
@@ -18,6 +18,14 @@ def GammaEntropy(shape, rate):
     return np.sum(shape - np.log(rate) + asp.special.gammaln(shape) + \
                   (1 - shape) * asp.special.digamma(shape))
 
+def DirichletEntropy(alpha):
+    sum_alpha = np.sum(alpha)
+    log_beta = np.sum(asp.special.gammaln(alpha)) \
+                - asp.special.gammaln(sum_alpha)
+    return log_beta - (np.shape(alpha) - sum_alpha) * asp.special.digamma(sum_alpha) \
+            - np.dot((alpha - 1), asp.special.digamma(alpha))
```

Comment:
If you ever call this with a 2d vector alpha (i.e. one with shape [k, 1]) it will fail unnecessarily.  Use ```len``` instead of ```shape```.